setup.py

```diff
@@ -15,6 +15,27 @@
 exec(compile(open(mod).read(), mod, 'exec'), {}, pkg)
 
 
+def parse_requirements(requirements, ignore=('setuptools',)):
+    """Read dependencies from requirements file and strip off version numbers.
+
+    Note: this imlementation does not support requirements files with extra
+    requirements (in square brackets) or '>=' and '<=' version qualifiers.
+
+    """
+    with open(requirements) as f:
+        packages = set()
+        for line in f:
+            line = line.strip()
+            if line.startswith(('#', '-r', '--')):
+                continue
+            if '#egg=' in line:
+                line = line.split('#egg=')[1]
+            pkg = line.split('==')[0].strip()
```

Comment:
But if being inclusive isn't accurate, are we really being inclusive? An example might be with Click which I understand has made some backwards incompatible changes. If we say click>=6, and version 7 comes out and horks up a lot of the features that we use in Watson... Watson is still going to need an upgrade.

It seems like there are three scenarios: 

1. Do minimum version pinning, and none of our dependencies ever make backwards incompatible changes.
2. Do minimum version pinning, and our dependencies **do**, in fact, make backwards incompatible changes, requiring users to upgrade Watson.
3. Do minimum and maximum version pinning, so when people try to install Watson they can't install it because they have too-new versions of some library. Which requires us to fix Watson and users to upgrade.

My personal preference is that things fail at the earliest opportunity. I don't want to install an application and try to use it only to find out I've got the wrong version installed. I want to know *now* that I can't install it.

I'm not sure what would happen if you install an app with min/max version pinning and then later try to upgrade a dependency. Would it fail, or just succeed and later cause issues with your app?